### PR TITLE
checkbox-group

### PR DESCRIPTION
**Pull request by AdityaVandan**
check-box-group (made from scratch)

### CODE REVIEW

File: src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+import { CheckBox, CHECKBOX_DIRECTION } from '../../atoms/CheckBox';
+
+import './checkBoxGroup.css';
+
+export { CHECKBOX_DIRECTION } from '../../atoms/CheckBox';
```
Comment (ankitchouhan1020) [inline src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx:7]: place it at the end of the file.

File: src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+const CheckBoxGroup = (props: Props) => {
+  const { containerClassName, checkBoxes, checkedList, onCheck } = props;
+
+  return (
+    <div className={`cbg29ParentDiv ${containerClassName}`}>
+      {checkBoxes.map((checkBox, index) => {
```
Comment (ankitchouhan1020) [inline src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx:14]: can we destructure checkbox here and spread rest of the props ?

```tsx
<CheckBox 
 isChecked={}
 {...rest}
/>

```

File: src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+type DefaultProps = {
+  containerClassName: string;
+}
+
+
+type RequiredProps = {
+  checkBoxes: CheckBox[];
+  checkedList: string[];
+  onCheck: (value: string, isChecked: boolean) => void;
```
Comment (ankitchouhan1020) [inline src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx:50]: param should be optional.

File: src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+
+type DefaultProps = {
+  containerClassName: string;
+}
+
+
+type RequiredProps = {
+  checkBoxes: CheckBox[];
```
Comment (ankitchouhan1020) [inline src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx:48]:
```ts
checkBoxes: { label: string, value: string}[]
```

File: src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
```
Comment (ankitchouhan1020) [inline src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx:1]: will discuss this later.

File: src/components/molecules/index.ts [hunk 6]
```diff
@@ -1 +1,4 @@
-export { }
+import '../../utils/styles/app.css';
+import '../../utils/styles/app.css';
```
Comment (ankitchouhan1020) [inline src/components/molecules/index.ts:2]: duplication

File: stories/CheckBoxGroup.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+import { Story } from "@storybook/react";
+import { action } from '@storybook/addon-actions';
+
+import { Props as CheckBoxGroupProps } from '../src/components/molecules/CheckBoxGroup/CheckBoxGroup';
+import { CheckBoxGroup, CHECKBOX_DIRECTION } from '../src/components/molecules';
+
+export default {
+  title: 'CheckBoxGroup',
+  component: CheckBoxGroup,
+  argTypes: {
```
Comment (ankitchouhan1020) [inline stories/CheckBoxGroup.stories.tsx:11]: remove if not required.

File: stories/CheckBoxGroup.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  }
+};
+
+const Template: Story<CheckBoxGroupProps> = (args) => {
+  const [checkedList, setCheckedList]: [string[], Function] = useState([]);
```
Comment (ankitchouhan1020) [inline stories/CheckBoxGroup.stories.tsx:16]:
```ts
[string[], Function] 
```
not necessary.

File: src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx (hunk 4 above)
Comment (AdityaVandan) [reply src/components/molecules/CheckBoxGroup/CheckBoxGroup.tsx:48]: actually just label and value are not enough

**Review by dheerajsingh123 — APPROVED** [review]

**Review by ankitchouhan1020 — APPROVED** [review]

**Pull request merged**